src/service/resources-impl.js

```diff
@@ -1723,7 +1723,7 @@ export class Resource {
       return Promise.resolve();
     }
 
-    if (!this.renderOutsideViewport() && !this.isInViewport()) {
+    if (!this.isInViewport() && !this.renderOutsideViewport()) {
```

Comment:
I'd rather not have `prerenderOutsideViewport`. Instead, ideally we'd do something like:
```
var r = x.rendersOutsideViewport();
var dist = abs(x.position - viewport);
if (dist <= r) {
  x.layout();
} else if (dist <= r * 2) {
  x.prerender();
} else {
  // do nothing at all.
}
```